### PR TITLE
3 add pagination in explorer

### PR DESCRIPTION
**Pull request by prllrp**
I think I managed to get the PR right this time, please let me know if it looks good.

As far as the feature,  I went ahead and created my own Pagination component. The default page size is 100  but  that can be easily changed or I can add a drop down for user input.

### CODE REVIEW

**Comment by calummoore** [comment]
You need to fix the lint issues. I'd recommend setting up vscode to fix these issues on save. https://www.digitalocean.com/community/tutorials/workflow-auto-eslinting

**Review by calummoore — CHANGES_REQUESTED** [review]

**Comment by calummoore** [comment]
Also every time I scroll down, the list goes to 0, which has the effect of scrolling me back to the top of the page. Either cache the results, or user the cursor approach we talked about.

**Comment by calummoore** [comment]
Also, can you rebase against main

**Comment by prllrp** [comment]
> Also every time I scroll down, the list goes to 0, which has the effect of scrolling me back to the top of the page. Either cache the results, or user the cursor approach we talked about.

I was only able to reproduce this once, and then without changing anything it went away. The list should never get set to 0. Please let me know if this persists for you

**Comment by calummoore** [comment]
Please mark the comments as resolved once you've completed them

**Comment by calummoore** [comment]
> I was only able to reproduce this once, and then without changing anything it went away. The list should never get set to 0. Please let me know if this persists for you

Yes, I'm still seeing this. This may be because when you change the query (e.g. ` .collection('Collection').limit(page * PAGE_LENGTH)`, the `data` prop becomes null while it's loading.

```js
const query = polybase
    .collection('Collection').limit(page * PAGE_LENGTH)

  const { data, loading, error } = useCollection<CollectionMeta>(
    pk
      ? query.where('publicKey', '==', pk)
      : q
```

**Comment by prllrp** [comment]
> 

Ok I'll implement the cursor tomorrow then, should fix that up.

**Comment by calummoore** [comment]
> > 
> 
> Ok I'll implement the cursor tomorrow then, should fix that up.

Yeah, or you can implement a cache in the pagination that when it receives null it doesn't update the value:

```js
const ref = useRef([])
if (items) {
  ref.current = items
}
```

File: src/features/collections/CollectionList.tsx [hunk 1]
```diff
@@ -33,12 +38,22 @@ export function CollectionList ({ pk }: CollectionListProps) {
...
 
+  const ref:any = useRef([])
+  if (items){
+    ref.current = items
+  }
+
+  const pageProps: PaginationProps = {
+    page: page || 1,
+    setPage,
+    pageLength:PAGE_LENGTH,
```
Comment (calummoore) [inline src/features/collections/CollectionList.tsx:49]: Can you add an eslint rule for this

**Review by calummoore — APPROVED** [review]

**Comment by calummoore** [comment]
Please can you rebase against main, and then I will merge. 👍

**Comment by calummoore** [comment]
And can you please add pagination to the CollectionListData view too

**Comment by prllrp** [comment]
> And can you please add pagination to the CollectionListData view too

Just to clarify, you mean on the table when you view a collection right? I saw there was an infinite scroll component on the table. Should I look into that?

**Comment by calummoore** [comment]
Please don't merge these in yourself, I'm not sure why it let you do that.

**Pull request merged**